### PR TITLE
Rely on subgraph for trending projects calculation

### PR DESCRIPTION
**Pull request by peripheralist**
## What does this PR do and why?

Updates the UI to rely on the subgraph for calculating trending projects. 
- Removes now-unused ipfsCache types, functions, etc
- Replaces trending projects calculation with simple graph query
- Removes `TrendingProject` type
- Adds properties to `Project` entity type:
```
  trendingPaymentsCount: number
  trendingScore: BigNumber
  trendingVolume: BigNumber
  createdWithinTrendingWindow: boolean
```

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [x] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/constants/trendingWindowDays.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const trendingWindowDays = 7
```
Comment (peripheralist) [inline src/constants/trendingWindowDays.ts:1]: might not be necessary. this value is only used in one place

File: src/models/subgraph-entities/vX/project.ts [hunk 2]
```diff
@@ -42,8 +42,13 @@ import { parseTapEventJson, TapEvent, TapEventJson } from '../v1/tap-event'
 type BaseProject = {
```
Comment (peripheralist) [inline src/models/subgraph-entities/vX/project.ts:42]: this file is 
- adding trending properties to Project type
- remove TrendingProject type
- bit of cleanup

File: src/models/subgraph-entities/vX/protocol-log.ts [hunk 3]
```diff
@@ -8,6 +8,8 @@ export type ProtocolLog = {
   paymentsCount: number
   redeemCount: number
   erc20Count: number
+  trendingLastUpdatedTimestamp: number
+  oldestTrendingPayEvent: string
```
Comment (peripheralist) [inline src/models/subgraph-entities/vX/protocol-log.ts:12]: these are primarily helper properties used by the subgraph itself. probably not needed by the UI

File: src/hooks/Projects.ts [hunk 4]
```diff
@@ -150,168 +139,20 @@ export function useProjectsSearch(handle: string | undefined) {
   )
 }
 
-// Returns projects with highest % volume increase in last week
```
Comment (peripheralist) [inline src/hooks/Projects.ts:153]: byeeeeeeee 👋

**Comment by peripheralist** [comment]
**DO NOT MERGE** until upcoming subgraph migration has completed.

File: src/models/subgraph-entities/vX/project.ts [hunk 5]
```diff
@@ -42,8 +42,13 @@ import { parseTapEventJson, TapEvent, TapEventJson } from '../v1/tap-event'
...
   id: string
   projectId: number
+  cv: CV
   owner: string
   createdAt: number
+  trendingPaymentsCount: number
```
Comment (johnnyd-eth) [inline src/models/subgraph-entities/vX/project.ts:48]: **Love**

**Comment by johnnyd-eth** [comment]
Sorry for getting to this so late @peripheralist ! All makes sense to me, and code looks very clean. I'll approve and leave the new subgraph deployment and merge of this one to you :)

**Comment by johnnyd-eth** [comment]
APPROVED (can't approve a drafted PR)

**Comment by peripheralist** [comment]
@johnnyd-eth thank you ser 🙏

Subgraph migration pending

**Pull request merged**